core/ledger/pvtdatastorage/store.go

```diff
@@ -1053,3 +1087,83 @@ func (c *collElgProcSync) done() {
 func (c *collElgProcSync) waitForDone() {
 	<-c.procComplete
 }
+
+type purgeUpdatesProcessor struct {
+	db           *leveldbhelper.DBHandle
+	maxBatchSize int
+
+	pvtWrites map[string]*rwsetutil.CollPvtRwSet
+	batch     *leveldbhelper.UpdateBatch
+
+	currentSize int
+}
+
+func newPurgeUpdatesProcessor(db *leveldbhelper.DBHandle, maxBatchSize int) *purgeUpdatesProcessor {
+	return &purgeUpdatesProcessor{
+		db:        db,
+		pvtWrites: map[string]*rwsetutil.CollPvtRwSet{},
+		batch:     db.NewUpdateBatch(),
+	}
+}
+
+func (p *purgeUpdatesProcessor) process(hashedIndexKey, hashedIndexVal []byte) error {
```

Comment:
Add a function comment explaining what the hashedIndexKey (purge height for a key) and hashedIndexVal represent, and what exactly this function will do. That is, it will find the referenced private data store height entry, remove the private key write from that height entry, and then re-save the height entry, and then also delete the HashedIndex entry itself.

Reply:
done